front/ui/ui-charts/src/spaceTimeChart/components/PathLayer.tsx

```diff
@@ -106,28 +106,39 @@ export const PathLayer = ({
   border,
 }: PathLayerProps) => {
   /**
-   * This function returns the list of points to join to draw the path. It will be both used to
-   * render the visible path, and the segments on the picking layer.
+   * This function returns the list of points to join to draw the path. As it can be discontinuous,
+   * it is returned as a Point[][]. For now, the only case for discontinuous paths is when the path
+   * stops on a flat step (in which case, we assume the path will be drawn differently in the flat
+   * step layer).
+   *
+   * It will be both used to render the visible path, and the segments on the picking layer.
    */
-  const getPathSegments = useCallback(
+  const getPathLines = useCallback(
     ({
       getTimePixel,
       getSpacePixel,
       spaceScaleTree,
+      flatSteps,
       timeAxis,
       spaceAxis,
-    }: SpaceTimeChartContextType): Point[] => {
-      const res: Point[] = [];
-      path.points.forEach(({ position, time }, i, a) => {
+    }: SpaceTimeChartContextType): Point[][] => {
+      const lines: Point[][] = [];
+      let line: Point[] = [];
+      const { points } = path;
+
+      for (let i = 0; i < points.length; i++) {
+        const { position, time } = points[i];
+
         if (!i) {
```

Comment:
nit : Could we put that instead ?
```suggestion
        if (i === 0) {
```